### PR TITLE
Feature/ng after view checked

### PR DESCRIPTION
**Pull request by garyluu**
This fixes (or works-around) the ExpressionChangedAfterItHasBeenCheckedError errors that we kept getting when enableProd() is removed ga4gh/dockstore#853:

Additionally, several other things were changed:
- all bodybuilder query building functions have been moved to its own query-builder service, no other service or component should be using bodybuilder
- search.component default test have been enabled
- default query-builder test is added
- search table (along with the tag cloud) has been moved to the search-results component

In the future, tag cloud should be its own component.  And the sidebar should also be its own component.

### CODE REVIEW

**Comment by garyluu** [comment]
wow, that's a pretty huge coverage decrease.

File: src/app/search/query-builder.service.ts [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+/**
+ * This service constructs all the querys and should be only class that interacts with the bodybuilder library.
+ *
+ * @export
+ * @class QueryBuilderService
+ */
+@Injectable()
+export class QueryBuilderService {
+    private shard_size = 10000;
```
Comment (denis-yuen) [inline src/app/search/query-builder.service.ts]: A comment why the shard size was chosen to be 10,000 would be useful here

File: src/app/search/search-results/search-results.component.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  // Tells the search service to tell the search filters to save its data
+  saveSearchFilter() {
+    this.searchService.toSaveSearch$.next(true);
+  }
+
+  tagClicked(clicked: CloudData) {
+    this.searchService.searchTerm$.next(true);
+    this.searchService.values$.next(clicked.text);
+    this.searchService.tagClicked$.next(true);
+    // this.searchTerm = true;
```
Comment (denis-yuen) [inline src/app/search/search-results/search-results.component.ts]: Delete this or put it in a descriptive if/else

File: src/app/search/search.component.ts [hunk 3]
```diff
@@ -1,23 +1,22 @@
...
+import { AdvancedSearchService } from './advancedsearch/advanced-search.service';
+import { ELASTIC_SEARCH_CLIENT } from './elastic-search-client';
+import { QueryBuilderService } from './query-builder.service';
+import { SearchService } from './search.service';
 
 /** TODO: ExpressionChangedAfterItHasBeenCheckedError is indicator that something is wrong with the bindings,
  *  so you shouldn't just dismiss it, but try to figure out why it's happening...
  **/
-enableProdMode();
+// enableProdMode();
```
Comment (denis-yuen) [inline src/app/search/search.component.ts]: Same here

**Pull request merged**